Develop

MartinLazarovBG opened the pull request.

MartinLazarovBG committed:
Fix more small bugs

MartinLazarovBG committed:
Fix more small bugs (#27)

MartinLazarovBG committed:
Fix table id persist

MartinLazarovBG committed:
Bugfix/fix small bugs (#29)

TheUntoldFable commented on src/app/cart/page.tsx:
```diff
@@ -37,6 +37,7 @@ export default function CartPage() {
 
     // Check if this restaurant uses pre-payment
     const isPrePayMode = restaurantData?.paymentInAdvance ?? false
+    console.log(restaurantInfo)
```
Не ни трябва :D

TheUntoldFable reviewed (changes requested):
Merge-ни си main, преди да го слееш.

MartinLazarovBG committed:
Fix redirect

MartinLazarovBG committed:
Fix/redirect (#30)

MartinLazarovBG committed:
Merge branch 'main' into chore/merge-main

MartinLazarovBG committed:
Chore/merge main (#31)

Commit:
some translations

Commit:
forgot to update couple of files related to order

balevdev committed:
Translations following current Architecture  (#32)

- Language Switcher Component
(src/components/common/language-switcher.tsx:1)
    - Dropdown menu with BG/EN language options
    - Integrated into the header, replacing the previous food icon
- Automatically invalidates all React Query caches on language change

  State Management

    - Language Store (src/store/language.ts:1)
    - Recoil atom for managing selected language
    - LocalStorage persistence using recoil-persist
    - Language mapping utilities (BG → "bulgarian", EN → "english")

  API Integration

    - Global Axios Interceptor (src/api/config.ts:31)
    - Automatically appends lang parameter to all API requests
    - Reads language from localStorage
    - Defaults to Bulgarian (BG) if not set

  Updated Hooks

All data fetching hooks now include language as a dependency and
parameter:

Commit:
i18 for nextJS

MartinLazarovBG committed:
fix ui issues

MartinLazarovBG committed:
fix total price formatting

balevdev committed:
i18 for nextJS (#33)

  What:
- Added complete internationalization (i18n) support to the Next.js
application with Bulgarian and English translations
- Integrated next-intl library with translation files for all
user-facing strings across navigation, cart, payment, and product
  components
- Replaced all hardcoded Bulgarian text strings throughout the
application with dynamic translation keys

  Why:
- Enable multi-language support to serve both Bulgarian and
English-speaking customers in the restaurant ordering system
- Improve maintainability by centralizing all text content in structured
JSON translation files instead of scattered strings
- Prepare the application for future expansion to additional languages
and international markets

  How:
- Installed next-intl library and configured IntlProvider wrapper in the
app layout with locale detection
- Created comprehensive translation dictionaries (messages/bg.json and
messages/en.json) covering all UI text across navigation, cart,
  orders, payments, tips, and error messages
- Refactored all components to use useTranslations hooks, replacing
hardcoded strings with translation keys (e.g., "Избрано" →
  tCart("selected"))

balevdev committed:
fix ui issues (#34)

uzun0ff committed:
[feat]: Adds rating and feedback form.

uzun0ff committed:
Merge branch 'develop' into feat/reting

balevdev committed:
[feat]: Adds rating and feedback form. (#35)

Commit:
next 15

Commit:
initial upgrade

Commit:
fix eslint compatibility

balevdev committed:
Feature/upgrade next and react (#36)

   - Upgrades Next.js from 14.x to **16.0.7**
   - Upgrades React from 18.x to **19.2.1**
   - Upgrades Tailwind CSS to **v4** with new PostCSS configuration
   - Updates Zustand to **v5** (replacing Recoil)
   - Updates all Radix UI components to latest versions

   ## Breaking Changes Addressed

- Migrated from `tailwind.config.ts` to Tailwind v4's CSS-based
configuration
   - Updated `postcss.config.js` to `postcss.config.mjs` for ESM support
   - Replaced `require()` with ES module imports in API routes
   - Updated async params handling in Next.js 16 page components

   ## Linting & Code Quality

   Many files were modified due to:
- **Prettier auto-formatting**: Tailwind class sorting (via
`prettier-plugin-tailwindcss`)
   - **ESLint fixes**: Removed unused imports and variables
- **Type safety improvements**: Replaced `any` types with proper types
   - **Code style**: Fixed duplicate imports, unused expressions

   ## Files Changed

   ~75 files modified, primarily:
   - Configuration files (package.json, tsconfig.json, eslint, postcss)
   - All page and component files (linting/formatting)
   - Store files (Zustand v5 migration)
   - Hook files (import organization)

   ## Test plan

   - [x] Build passes (`npm run build`)
   - [x] ESLint passes with no errors
   - [x] Manual testing of all pages
   - [x] Verify payment flow works
   - [x] Test internationalization

uzun0ff committed:
[hotfix]: Update card, remove card from .gitIgnore

uzun0ff committed:
[fix]: ignore env

uzun0ff committed:
[fix]: Uses EUR for main currency, updates socket error handling, removes .env from git history, changes stripe currency to eur

uzun0ff committed:
[chore]: Add english translation

borislawww committed:
[feat]: EUR as main currency (#37)

uzun0ff committed:
[feat]: Adds rating for products

MartinLazarovBG committed:
Fix prices

MartinLazarovBG committed:
Fix error

MartinLazarovBG committed:
Fix build

borislawww committed:
[Feature]: Adds rating for products when finishing order (#38)

MartinLazarovBG committed:
Fix price

MartinLazarovBG committed:
Fix/small bugs (#39)
